version: "3.8"

services:
  gh_fastapi:
    container_name: gh_fastapi
    build:
      context: ./
      dockerfile: ./Dockerfile
    volumes:
      - .:/app
      - ${GRAPH_CACHE_DIR}:/graph_cache
    expose:
      - ${FASTAPI_PORT}
    labels:
      - traefik.enable=true
      - traefik.http.routers.app-http.rule=Host(`${DOMAIN_NAME}`)
      - traefik.http.routers.app-http.tls=true
      - traefik.http.routers.app-http.tls.certresolver=letsencrypt

  traefik:
    build:
      context: .
      dockerfile: Dockerfile.traefik
    command:
      - "--certificatesresolvers.myresolver.acme.email=${ERROR_EMAIL}"
      - "--certificatesresolvers.myresolver.acme.storage=/certificates/acme.json"
    ports:
      - 80:80
      - 443:443
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./traefik-public-certificates:/certificates"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.dashboard.rule=Host(`dashboard.${DOMAIN_NAME}`) && (PathPrefix(`/`))"
      - "traefik.http.routers.dashboard.tls=true"
      - "traefik.http.routers.dashboard.tls.certresolver=letsencrypt"
      - "traefik.http.routers.dashboard.service=api@internal"
      - "traefik.http.routers.dashboard.middlewares=auth"
      - "traefik.http.middlewares.auth.basicauth.users=${DASHBOARD_USERNAME}:${DASHBOARD_PASSWORD}"
